pilot/cmd/pilot-discovery/main.go

```diff
@@ -76,6 +90,24 @@ var (
 				return fmt.Errorf("failed to start discovery service: %v", err)
 			}
 
+			if serverArgs.Chiron.EnableChiron {
+				opts := &serverArgs.Chiron
+				k8sClient, err := kubelib.CreateClientset(serverArgs.Config.KubeConfig, "")
```

Comment:
Pilot already creates a k8s client, please reuse. I don't think it's efficient to create multiple.

Reply:
Done.